f600233d1b154d507b782f84604bcb12/update/sys_script_include_d393df0197141110c351f5b3f153af5d.xml

```diff
@@ -8,22 +8,106 @@
         <description>Logs payload from EDA PDS </description>
         <name>edaPdsResult</name>
         <script><![CDATA[function edaPdsResult(http_payload){
-	gs.info(http_payload.getBody());
-	gs.info(http_payload.getStatusCode());
-	gs.info(http_payload.getAllHeaders());
-	
+	try{
+		// Log http_payload
+		gs.info(http_payload.body);
+		gs.info(http_payload.status_code);
+		gs.info(http_payload.headers);
+
+		// Validate the XML before converting it to a string
+		// validateXML will return null if it passes validation
+		var xmlBody = http_payload.body;
+		var xmlValid = GlideXMLUtil.validateXML(xmlBody,false, false);
+		if (xmlValid == null){
+			// Parse the XML string as an XMLDocument2()
+			// This is required to use gs.xmlToJSON() without errors
+			// For more information please see: https://support.servicenow.com/kb?id=kb_article_view&sysparm_article=KB0784264
+			var xmlStringBody = xmlBody.toString();
+			var xmlDoc = new XMLDocument2();
+			xmlDoc.setNamespaceAware(true);
+			xmlDoc.parseXML(xmlStringBody);
+			var jsonObject = gs.xmlToJSON(xmlDoc);
+
+			// As a result of parseXML, we are left with LineItems as an Array or an Object
+			// If it is an Array, that means there are CLIN records to create
+			// If it is an Object, that means there is a single CLIN record instead of a CLIN and INFOSLIN, requiring no CLIN creation
+			var lineItems = jsonObject.ProcurementDocument.AwardInstrument.ContractLineItems.LineItems;
+			var clins = [];
+			gs.info(Array.isArray(lineItems));
+			if (Array.isArray(lineItems)){
+				for (var i = 0; i< lineItems.length; i++){
```

Comment:
No need to change, but just for future ref I believe you can use `.map()`, `.filter()`, and `.forEach()` but you have to use `function() {}` instead of arrow functions. Test out in a background script before doing though.